clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -178,11 +179,13 @@ public synchronized boolean updateLastSeenEpochIfNewer(TopicPartition topicParti
      * @param topicPartition       topic+partition to update the epoch for
      * @param epoch                the new epoch
      * @param epochTest            a predicate to determine if the old epoch should be replaced
+     * @param overwriteNullEpoch   if we should allow overwriting a missing epoch
      * @return true if the epoch was updated, false otherwise
      */
     private synchronized boolean updateLastSeenEpoch(TopicPartition topicPartition,
                                                      int epoch,
-                                                     Predicate<Integer> epochTest) {
+                                                     Predicate<Integer> epochTest,
+                                                     boolean overwriteNullEpoch) {
```

Comment:
This seems unused?

Reply:
Yea, sorry i mistakenly left out the actual changes in this commit. Will push with the tests shortly.